### PR TITLE
Move CodeMirror to webui2

### PR DESCRIPTION
**Pull request by saraycp**
- Changes slightly the CSS of the editor's toolbar.
- Refactors the editor code.
- Creates a test helper, `fill_in_editor_field`, to easily write text
  inside CodeMirror editor from feature tests.

Co-authored-by: Björn Geuken <bgeuken@suse.de>

### CODE REVIEW

File: src/api/app/assets/javascripts/webui2/cm2/use-codemirror.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  editor.id = id;
+  if (!read_only) {
+    editor.setSelections(editor);
+
+    editor.on('change', function (cm) {
+      var changed = true;
+    cm.updateHistory(cm);
+    if (cm.historySize().undo > 0)
+      $("#save_" + id).removeClass('inactive');
+      else
```
Comment (bgeuken) [inline src/api/app/assets/javascripts/webui2/cm2/use-codemirror.js]: The indentation of the previous 3 lines looks wrong

**Comment by dmarcoux** [comment]
Forget my previous comment, it's working... Sorry! I didn't include this line at the top of a Bootstrap view using Code Mirror:

`- content_for(:content_for_head, javascript_include_tag('webui/application/cm2/index'))`

Here's the result:

code-mirror

**Comment by saraycp** [comment]
@dmarcoux  no worries, while waiting for you to check that, we improved the code a bit :)

@openSUSE/obs-frontend could you please review again? Also take into account for the rest of the views using CodeMirror that you have to include the javascript file in the view itself. Something like:
`- content_for(:content_for_head, javascript_include_tag('webui/application/cm2/xxx))`

If you have a better way to do this without including it on each file, please share! :bouquet:

**Comment by saraycp** [comment]
[Code Mirror](https://trello.com/c/EIBpUT8w/864-code-mirror)

**Pull request merged**